scripts/performance/benchmark-download

```diff
@@ -0,0 +1,171 @@
+import os
+import tempfile
+import shutil
+import subprocess
+
+from botocore.session import get_session
+from s3transfer.manager import TransferManager
+
+
+TEMP_FILE = 'temp'
+TEMP_KEY = 'temp'
+KB = 1024
+SIZE_SUFFIX = {
+    'kb': 1024,
+    'mb': 1024 ** 2,
+    'gb': 1024 ** 3,
+    'tb': 1024 ** 4,
+    'kib': 1024,
+    'mib': 1024 ** 2,
+    'gib': 1024 ** 3,
+    'tib': 1024 ** 4,
+}
+
+
+def human_readable_to_bytes(value):
+    """Converts a human readable size to bytes.
+    :param value: A string such as "10MB".  If a suffix is not included,
+        then the value is assumed to be an integer representing the size
+        in bytes.
+    :returns: The converted value in bytes as an integer
+    """
+    value = value.lower()
+    if value[-2:] == 'ib':
+        # Assume IEC suffix.
+        suffix = value[-3:].lower()
+    else:
+        suffix = value[-2:].lower()
+    has_size_identifier = (
+        len(value) >= 2 and suffix in SIZE_SUFFIX)
+    if not has_size_identifier:
+        try:
+            return int(value)
+        except ValueError:
+            raise ValueError("Invalid size value: %s" % value)
+    else:
+        multiplier = SIZE_SUFFIX[suffix]
+        return int(value[:-len(suffix)]) * multiplier
+
+
+def create_file(filename, file_size):
+    with open(filename, 'wb') as f:
+        for i in range(0, file_size, KB):
+            f.write(b'a' * i)
+
+
+def benchmark_download(args):
+    # Create a temporary directory to use for scratch work.
+    tempdir = tempfile.mkdtemp()
+    temp_file = os.path.join(tempdir, TEMP_FILE)
+
+    # Override the name of the temp file if a target was given.
+    if args.target_download:
+        temp_file = os.path.abspath(
+            os.path.expanduser(args.target_download))
+    session = get_session()
+    client = session.create_client('s3')
+    s3_key = args.existing_s3_key
+    try:
+        # If an existing s3 key was not specified, then create a temporary
+        # file of that size for the user and upload it.
+        if not args.existing_s3_key:
+            # Create the temporary file.
+            create_file(temp_file, args.file_size)
+
+            # Create the temporary s3 key
+            s3_key = TEMP_KEY
+            upload_file(client, temp_file, args.s3_bucket)
+
+        # Glue together the download-file script that is used in the benchmark
+        # script.
+        download_file_script = (
+            './download-file --file-name %s --file-type %s --s3-bucket %s '
+            '--s3-key %s' % (
+                temp_file, args.file_type, args.s3_bucket, s3_key)
+        )
+        # Add in the output-file for benchmark if it was specified.
+        benchmark_args = ['./benchmark', download_file_script]
+        if args.output_file:
+            benchmark_args.extend(['--output-file', args.output_file])
+        # Run the benchmark script now.
+        subprocess.check_call(benchmark_args)
+    finally:
+        # Remove the scratch directory.
```

Comment:
Same thing as before.  These comments are distracting.  I think it's reasonable to assume a python person will know what `shutil.rmtree` does.